### Sales Performance Funnel Analysis for a Multi-Store Supermarket Chain

###### Business Objective:

* Understand how orders flow from all sales - discounted sales - high discount sales - profitable sales
* Identify profit leakages due to discounts
* Provide recommendations for pricing and discount strategies

###### Funnel Stages:

* Total Orders – All transactions
* Discounted Orders – Orders where Discount > 0
* High Discount Orders – Orders where Discount >= 20%
* Profitable Orders – Orders where Profit > 0

In [10]:
# import libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [2]:
# import dataset

df = pd.read_csv("SampleSuperstore.csv")

In [3]:
# view head

df.head()

Ship Mode    Segment        Country             City       State  \
0    Second Class   Consumer  United States        Henderson    Kentucky   
1    Second Class   Consumer  United States        Henderson    Kentucky   
2    Second Class  Corporate  United States      Los Angeles  California   
3  Standard Class   Consumer  United States  Fort Lauderdale     Florida   
4  Standard Class   Consumer  United States  Fort Lauderdale     Florida   

   Postal Code Region         Category Sub-Category     Sales  Quantity  \
0        42420  South        Furniture    Bookcases  261.9600         2   
1        42420  South        Furniture       Chairs  731.9400         3   
2        90036   West  Office Supplies       Labels   14.6200         2   
3        33311  South        Furniture       Tables  957.5775         5   
4        33311  South  Office Supplies      Storage   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164

In [4]:
# view dataset info

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Ship Mode     9994 non-null   object 
 1   Segment       9994 non-null   object 
 2   Country       9994 non-null   object 
 3   City          9994 non-null   object 
 4   State         9994 non-null   object 
 5   Postal Code   9994 non-null   int64  
 6   Region        9994 non-null   object 
 7   Category      9994 non-null   object 
 8   Sub-Category  9994 non-null   object 
 9   Sales         9994 non-null   float64
 10  Quantity      9994 non-null   int64  
 11  Discount      9994 non-null   float64
 12  Profit        9994 non-null   float64
dtypes: float64(3), int64(2), object(8)
memory usage: 1015.1+ KB


###### Data preparation

In [6]:
# Ensure proper types (already clean)
df['Discount'] = df['Discount'].astype(float)
df['Sales'] = df['Sales'].astype(float)
df['Profit'] = df['Profit'].astype(float)

###### Defining Funnel and Stages

In [14]:
profitable_high_discount = df[
    (df['Discount'] >= 0.2) & (df['Profit'] > 0)
].shape[0]

funnel_df = pd.DataFrame({
    'Stage': [
        'Total Orders',
        'Discounted Orders',
        'High Discount Orders',
        'Profitable High Discount Orders'
    ],
    'Orders': [
        total_orders,
        discounted_orders,
        high_discount_orders,
        profitable_high_discount
    ]
})

# Conversion rate
conversion_rates = [100]
for i in range(1, len(funnel_df)):
    rate = (funnel_df.loc[i, 'Orders'] / funnel_df.loc[i-1, 'Orders']) * 100
    conversion_rates.append(rate)

funnel_df['Conversion_Rate (%)'] = conversion_rates
funnel_df


Stage  Orders  Conversion_Rate (%)
0                     Total Orders    9994           100.000000
1                Discounted Orders    5196            51.991195
2             High Discount Orders    5050            97.190146
3  Profitable High Discount Orders    3165            62.673267

In [15]:
# visualizing funnel

fig = px.funnel(funnel_df, x='Orders', y='Stage', title='Sales Funnel Analysis')
fig.show()

###### Business Interpretation (Very Important)
###### Total Orders ---> Discounted Orders (52%)

* About half of all orders receive discounts
* Indicates heavy reliance on discounting to drive sales
* Potential risk to long-term profitability

######  Discounted Orders ---> High Discount Orders (97%)

* Nearly all discounted orders have high discounts (≥20%)
* Suggests lack of tiered or controlled discount strategy
* Discounts are not optimized — most are aggressive

###### High Discount Orders ---> Profitable High Discount Orders (62.7%)

* ~37% of high-discount orders are unprofitable
* High discounts significantly erode margins
* Indicates opportunity to:
* Cap discounts
* Apply discounts selectively by category or region

######  Key Drop-Off Point (Critical Insight)

* Biggest leakage occurs at the final stage
* High discounts do not consistently translate into profitable sales